include/inc_front/front.func.inc.php

```diff
@@ -1541,6 +1552,164 @@ function html_parse_idlink($matches) {
 	$replace .= '>' . $matches[2] . '</a>';
 	return $replace;
 }
+/*
+ * Parse a aliaslink, output: a link to intern article (in case of unknown article: fallback to linktext)
+ * @param: array(match,id/alias,linktext)
+ * @author: q23/geckse
+ */
+function html_parse_aliaslink($para) {
+	
+	if(!is_array($para)) return "";
+	
+	$id = 0;
+	$alias = "";
+	if(isset($para[2])){
+		$text = $para[2];
+	} else {
+		$text = "";	
+	}
+	
+	// prepared fallback
+	$link = $text;
+	
+	// clean $par 1
+	$para[1] = trim(str_replace('&nbsp;', '', $para[1]));
+	$para[1] = str_replace(':', '', $para[1]);	
+	$para[1] = preg_replace('/\s+/', '', $para[1]);
+	
+	// got an anchor?
+	if(strpos($para[1], '#') !== false) {
+		list($para[1], $anchor) = explode('#', $para[1], 2);
+		if($anchor) {
+			$anchor = '#'.$anchor;
+		}
+	} else {
+		$anchor = '';
+	}
+	
+	// link or alias?
+	if(is_numeric(trim($para[1]))){
+		$id = intval(trim($para[1]));
+	} else {
+		$alias = trim($para[1]);		
+	}
+	
+	$sql  = "SELECT article_alias, article_title ";
+    $sql .= "FROM ".DB_PREPEND."phpwcms_article ";
+    if($id == 0){
+  	  $sql .= "WHERE article_alias='".$alias."' ";
+    } else {
+  	  $sql .= "WHERE article_id=".$id." ";	    
+    }    
+    $sql .= "AND article_aktiv=1 AND article_deleted=0 ";
+	if(!PREVIEW_MODE) {
+	    $sql .= "AND article_public=1 AND (article_archive_status=1 AND article_begin<NOW() ";
+		$sql .= "OR article_archive_status=0 AND article_end>NOW()) ";
+    }
+    $sql .= " LIMIT 1";
+    $result = _dbQuery($sql);
+ 	if(is_array($result) && isset($result[0])) $result = $result[0];   
+    
+    if(isset($result['article_alias'])){
+		if($text == ""){
```

Comment:
always try to be precise `===`, `!==`. Use single quotes instead of double quotes whenever it is enough. I know, the old style and lot of this in my code too.